flexibleengine/utils.go

```diff
@@ -21,12 +21,28 @@ type ErrorResp struct {
 	ErrorMsg  string `json:"error_msg"`
 }
 
-// ParseErrorMsg is used to unmarshal the error body to ErrorResp
-// usage: resp, pErr := ParseErrorMsg(err.Body)
-func ParseErrorMsg(body []byte) (ErrorResp, error) {
-	resp := ErrorResp{}
-	err := json.Unmarshal(body, &resp)
-	return resp, err
+// ParseErrorMap parses the error message from the API and returns a map
+func ParseErrorMsg(data interface{}) (ErrorResp, error) {
```

Comment:
@azrod the function is tested well, please keep it as is.

Reply:
I'm getting an error and not a []byte. If you know how to do I let you modify

Reply:
you can use it like this: `ParseErrorMsg(err.Body)`

Reply:
```
errResp, nErr := ParseErrorMsg(err.Body)
```

`err.Body undefined (type error has no field or method Body)`

Reply:
you should convert the error to golangsdk.ErrDefault4xx